Source/ACE/Entity/Player.cs

```diff
@@ -50,6 +50,12 @@ public sealed class Player : Creature, IPlayer
         /// </summary>
         private Dictionary<ObjectGuid, WorldObject> interactiveWorldObjects = new Dictionary<ObjectGuid, WorldObject>();
 
+        private Dictionary<ObjectGuid, WorldObject> vendorsLastKnownDefaultList = new Dictionary<ObjectGuid, WorldObject>();
+
+        private ObjectGuid currentVendorObjectGuid;
+
+        private double currentVendorSellRate = 1.0;
```

Comment:
Not a fan of this implementation.  This is a Vendor property for use in the Vendor class.  Putting it here is a major code smell for violation of proper encapsulation.